apps/sms/views/shared/js/utils.js

```diff
@@ -945,24 +945,33 @@
     },
 
     /**
-     * Returns Promise instance that is resolved instantly if document is
-     * visible (document.hidden === false), othewise it's resolved only once
-     * `visibilitychange` event is fired.
-     * @returns {Promise.<void>}
+     * Returns Promise instance that is resolved only once specified event is
+     * fired on the specified event target.
+     * @param {EventTarget} eventTarget Event target to listen event on.
+     * @param {String} eventName Name of the event to listen for.
+     * @returns {Promise.<*>} Promise resolved with event object if any.
      */
-    onceDocumentIsVisible() {
-      if (!document.hidden) {
-        return Promise.resolve();
-      }
-
+    onceEventIsFired(eventTarget, eventName) {
```

Comment:
Actually I'd have integrated the timeout inside "onceEventIsFired" :)

but anyway I agree with you, let's remove the support for this non-existant event. We can ressurect it later when it comes to life.

But I'm not so sure about replacing the "load" with our internal ready (in split views). The semantic is that "afterEnter" should happen after the transition happened, after the window is displayed. But maybe it works after all...